src/boot/efi/meson.build

```diff
@@ -218,18 +218,21 @@ if have_gnu_efi
 
         libgcc_file_name = run_command(efi_cc + ['-print-libgcc-file-name']).stdout().strip()
         systemd_boot_efi_name = 'systemd-boot@0@.efi'.format(EFI_MACHINE_TYPE_NAME)
+        stub_so_name = 'linux@0@.so.stub'.format(EFI_MACHINE_TYPE_NAME)
         stub_efi_name = 'linux@0@.efi.stub'.format(EFI_MACHINE_TYPE_NAME)
         no_undefined_symbols = find_program('no-undefined-symbols.sh')
 
-        foreach tuple : [['systemd_boot.so', systemd_boot_efi_name, systemd_boot_objects],
-                         ['stub.so', stub_efi_name, stub_objects]]
+        foreach tuple : [['systemd_boot.so', systemd_boot_efi_name, systemd_boot_objects, 'systemd_boot.so', false],
+                         ['stub.so', stub_efi_name, stub_objects, stub_so_name, true]]
                 so = custom_target(
                         tuple[0],
                         input : tuple[2],
-                        output : tuple[0],
+                        output : tuple[3],
```

Comment:
Hmm, so now the target name is different than the file that is written. I don't think that's useful. Why not just change the target name?

Reply:
because that breaks things. There is this `set_variable(tuple[0].underscorify() + '_stub', stub)` later down which then is used in the test_efi_disk_img as input.

I think that parses the tuple[0] and sets variable name for it. It seems very odd. But changing the target name, breaks the build of test_efi_disk_img.

Any suggestions on how to make this better are welcome, this is the only thing I could come up with that worked.

Reply:
Please push the latest version (with the "elf" change I guess…).